c/expr/reduceop.cc

```diff
@@ -41,19 +42,25 @@ static Column* reduce_first(const Column* arg, const Groupby& groupby) {
     return Column::new_data_column(arg->stype(), 0);
   }
   size_t ngrps = groupby.ngroups();
-  // groupby.offsets array has length `ngrps + 1` and contains offsets of the
-  // beginning of each group. We will take this array and reinterpret it as a
-  // RowIndex (taking only the first `ngrps` elements). Applying this rowindex
-  // to the column will produce the vector of first elements in that column.
-  arr32_t indices(ngrps, groupby.offsets_r());
+  arr32_t indices(ngrps);
```

Comment:
The original constructor `arr32_t indices(ngrps, groupby.offsets_r());` simply uses the pointer as borrowed; however `arr32_t indices(ngrps);` allocates new memory and then you need to copy data into that memory. This is very wasteful.

Reply:
Pushed implementation that doesn't involve `memcpy`. The segfault problem is quite interesting, for me it appears in release mode only. Will work on it.